docs/bicep.md

```diff
@@ -1,10 +1,169 @@
+
+Modules provide a structured way to organize and reuse infrastructure definitions. While modules enhance maintainability by breaking down complex deployments into manageable components, they also serve a critical function in switching execution context during a Bicep template deployment. See the section about [Cross-Subscription deployments](#cross-subscription-deployments) for more details.
+
+### Parameters
+
+The configuration directory contains Bicep parameter files (`.bicepparam`). Each top-level Bicep template has a corresponding Bicep parameter file to define its configuration. These parameter files are Go templates, which are processed dynamically to generate final configuration files based on the targeted cloud, deployment environment, and region. The configuration references used in these Bicep parameter Go templates adhere to the configuration management concept described in the [configuration documentation](configuration.md).
+
+```bicep
+using '../templates/region.bicep'
+
+param globalRegion = '{{ .global.region }}'
+param regionalRegion = '{{ .region }}'
+```
+
+## Deploying Bicep templates
+
+In ARO HCP, we deploy Bicep templates via [pipeline](pipeline.md) files by supporting a dedicated step type for ARM/Bicep deployments.
+
+> [!IMPORTANT]
+> Read the documentation about [pipeline files](pipeline-concept.md) and their general format and functionality. The following documentation covers only the Bicep specific information
+
+```yaml
+$schema: "pipeline.schema.v1"
+
+serviceGroup: Microsoft.Azure.ARO.HCP.Region
+rolloutName: Region Rollout
+resourceGroups:
+- name: {{ .regionRG }}                                        (1)
+  subscription: {{ .svc.subscription }}                        (2)
+  steps:
+  - name: region                                               (3)
+    action: ARM                                                (4)
+    template: templates/my-template.bicep                      (5)
+    parameters: configurations/my-template.tmpl.bicepparam     (6)
+    deploymentLevel: ResourceGroup/Subscription                (7)
+    variables:                                                 (8)
+      ...
+    [outputOnly: true/false]                                   (9)
+```
+
+1. The name of Azure resourcegroup targeted by this deployment
+2. The name of the Azue Subscription targeted by this deployment. When deploying via EV2, this needs to reference an [EV2 subscription key](https://ev2docs.azure.net/features/service-artifacts/actions/subscriptionProvisioningParameters.html#subscription-key)
+3. The name of the Azure deployment
+4. The action type `ARM` marks this step as ARM/Bicep deployment action
+5. File reference to the Bicep template, relative to the location of the pipeline file
+6. File reference to the Bicep parameter file, relative to the location of the pipeline file.
```

Comment:
Perhaps worth noting, this should be the template

Reply:
added